shellfoundry_traffic/script_utils.py

```diff
@@ -18,15 +18,26 @@
 
 class ScriptCommandExecutor:
 
-    def __init__(self, script_definition_yaml: str) -> None:
-        shell_definition_yaml = Path(os.getcwd()).joinpath(f'{script_definition_yaml}.yaml')
-        with open(shell_definition_yaml, 'r') as file:
+    def __init__(self, script_definition: str) -> None:
+        script_definition_yaml = (script_definition if script_definition.endswith('.yaml') else
+                                  f'{script_definition}.yaml')
+        script_definition_yaml_full_path = Path(os.getcwd()).joinpath(script_definition_yaml)
+        with open(script_definition_yaml_full_path, 'r') as file:
             self.script_definition = yaml.safe_load(file)
         self.dist = Path(os.getcwd()).joinpath('dist')
         self.script_zip = self.dist.joinpath(f'{self.script_definition["metadata"]["script_name"]}.zip')
 
     def should_zip(self, file: str) -> bool:
-        return file not in self.script_definition['files']['exclude']
+        """ Returns whether the file should be added to the shell zip file or not. """
+        if 'files' in self.script_definition.keys():
+            if self.script_definition['files'] != 'exclude':
+                return False
+            elif self.script_definition['files']['exclude'] is None:
+                return False
+            else:
+                return True
+        else:
+            return False
```

Comment:
If no files entry then there are no files to include so the files should be zipped.

Reply:
OK, this is in the right direction, but why not be positive:
if 'files' in self.script_definition:
    if 'exclude' in self.script_definition['files']:
        return file not in self.script_definition['files']['exclude']
return True

Note that I assume that if files/exclude exists it includes a non empty list of files to exclude. But as I said, I'm OK with this assumption.

Always prefer positive question. Use
if True
Rather than
if not False